src/number_format.js

```diff
@@ -362,20 +371,30 @@ class NumberFormat extends React.Component {
 
   /** methods to remove formattting **/
   removePrefixAndSuffix(val: string) {
-    const {format, prefix, suffix} = this.props;
+    const {format, prefix, suffix, accounting} = this.props;
 
     //remove prefix and suffix
     if (!format && val) {
-      const isNegative = val[0] === '-';
+      let isNegative = val[0] === '-';
+      let stringEnd = val.length;
+
+      if (accounting) {
+        isNegative = val[0] === '(';
+        stringEnd = val.length - 1;
+      }
 
       //remove negation sign
-      if (isNegative) val = val.substring(1, val.length);
+      if (isNegative) val = val.substring(1, stringEnd);
 
       //remove prefix
       val = prefix && val.indexOf(prefix) === 0 ? val.substring(prefix.length, val.length) : val;
 
       //remove suffix
-      const suffixLastIndex = val.lastIndexOf(suffix);
+      let suffixLastIndex = val.lastIndexOf(suffix);
+      if (accounting && isNegative) {
```

Comment:
You need to handle adding parentheses back. 
We do it on 
```js
if (isNegative) val = '-' + val;
```

Reply:
I avoided that approach so the number remains a valid floating point number when passed to inner functions.
The adding/removing logic for parentheses is in removePrefixAndSuffix and formatNumString.
I think this is the right approach so the underlying logic remains the same between the two, that way this setting just adds an automatic prefix/suffix to the number while removing the - when formatting the output.